functions/zoom-uploader.py

```diff
@@ -288,19 +281,43 @@ def ingest(self):
             ("spatial", (None, "Zoom"))
         ]
 
-        for s3_filename in video_files:
-            url = self._generate_presigned_url(s3_filename)
-            params.extend([
-                ("flavor", (None, escape(ZOOM_OPENCAST_FLAVOR))),
-                ("mediaUri", (None, url))
-            ])
+        file_params = self._generate_file_params()
+        if not len(file_params):
+            raise Exception("No recording file view types that we want")
+        else:
+            params.extend(file_params)
 
         resp = oc_api_request("POST", endpoint, files=params)
-
         logger.debug({"addMediaPackage": resp.text})
-
         self.workflow_xml = resp.text
 
+    def _generate_file_params(self):
+
+        file_params = []
+
+        def make_param_pair(view_type, flavor):
+            if view_type not in self.s3_filenames:
+                return []
+            for s3_file in self.s3_filenames[view_type]:
+                logger.info("Found '{}' view: {}".format(view_type, s3_file))
+                s3_url = self._generate_presigned_url(s3_file)
+                return [
+                    ("flavor", (None, escape(flavor))),
+                    ("mediaUri", (None, s3_url))
+                ]
```

Comment:
Per @rute-santos , the order matters. If there is no "active_speaker", then "shared_screen"  should be assigned flavor ""multipart/chunked+source", and gallery to "presentation/chunked+source". If only gallery, it should be assigned "multipart/chunked+source". i.e. this looks like a straight mapping loop.

Reply:
this was addressed as part of a general reworking